src/Rules/RequiredIf.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Oshomo\CsvUtils\Rules;
+
+use Oshomo\CsvUtils\Contracts\ArrayParameterizedRuleInterface;
+use Oshomo\CsvUtils\Contracts\ValidationRuleInterface;
+
+class RequiredIf implements ValidationRuleInterface, ArrayParameterizedRuleInterface
+{
+    /**
+     * Should return an array of the allowed parameters.
+     * The allowed parameters should be tokenized string
+     * e.g :min, :max, :first, :last etc.
+     */
+    public function allowedParameters(): array
+    {
+        // required_if validation uses a dependent field
+        // marked as :other_field and an array of values
+        // which are allowed for the dependent field marked
+        // as :other_values
+        return [':other_field', ':other_values'];
+    }
+
+    /**
+     * Should return an array of parameter values as strings
+     * parsed in the same order and format as allowedParameters().
+     * This will aid in mapping our parameters to their placeholders.
+     */
+    public function parseParameterValues(array $parameters): array
```

Comment:
I am not sure why this was needed at all. Maybe I am not seeing something. My guess is this will be used like `["requiredIf:field,value"]`

Reply:
Yes, for my use case I had a couple of fields which had to be made required in case some target column had one of the values specified in :other_values.

Reply:
I see. It seems we can't run away from supporting array parameters. Let's do the then. I will make a fix to support array parameters like so,  all rules can accept parameters in one of the following ways 

- `["requiredIf:field,value"]`
- `["requiredIf" => ["field" => "value"]`
- `["requiredIf" => ["field1" => "value1", "field2" => "value2"]`
- `["requiredIf" => ["field1" => ["value1", "value2"]]`

What do you think? I will try to push this over the weekend.

Reply:
Sounds good.